src/EditorFeatures/CSharpTest/ConvertForToForEach/ConvertForToForEachTests.cs

```diff
@@ -1072,11 +1072,11 @@ void Test(MyList list)
 
 class MyList
 {
-  public object this[int i] { get; }
+    public object this[int i] { get => default; }
 
-  public Enumerator GetEnumerator() { }
+    public Enumerator GetEnumerator() { return default; }
 
-  public struct Enumerator { public object Current { get; } }
+    public struct Enumerator { public object Current { get; } public bool MoveNext() => true; }
```

Comment:
needed to make this change, otherwise the compiler felt that the type didn't properly match the 'foreach' pattern, and didn't want to consider hte types the same for reduction.